### PR TITLE
feat: implicit group auth fields in selection sets

### PR DESCRIPTION
**Pull request by svidgen**
#### Description of changes

Adds implied group fields from auth rules to selection sets. (Implicit owner fields were already being included.)

#### Description of how you validated changes

Added models and tests.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api-graphql/__tests__/internals/generateClient.test.ts [hunk 1]
```diff
@@ -3,13 +3,15 @@ import { Amplify, AmplifyClassV6 } from '@aws-amplify/core';
 import { generateClient } from '../../src/internals';
 import configFixture from '../fixtures/modeled/amplifyconfiguration';
 import { Schema } from '../fixtures/modeled/schema';
+import { Observable, from } from 'rxjs';
+import * as internals from '../../src/internals';
```
Comment (svidgen) [inline packages/api-graphql/__tests__/internals/generateClient.test.ts:7]: Hmm ... I think this import was a mistake. I'll take a look and rev or resolve.

File: packages/api-graphql/__tests__/internals/generateClient.test.ts [hunk 2]
```diff
@@ -62,47 +64,6 @@ function makeAppSyncStreams() {
 	return { streams, spy };
 }
 
-/**
```
Comment (svidgen) [inline packages/api-graphql/__tests__/internals/generateClient.test.ts:65]: Moved to https://github.com/aws-amplify/amplify-js/pull/13161/files#diff-10fce1a178e4b14f1b3e94a50363fc35edb027d8889e55f4e1d974f7f840a55eR3

File: packages/api-graphql/__tests__/internals/implicit-auth-fields.test.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	updatedAt: new Date().toISOString(),
+};
+
+/**
+ *
+ * @param value Value to be returned. Will be `awaited`, and can
+ * therefore be a simple JSON value or a `Promise`.
+ * @returns
+ */
+function mockApiResponse(value: any) {
```
Comment (svidgen) [inline packages/api-graphql/__tests__/internals/implicit-auth-fields.test.ts:32]: This one seemed both small and easier to duplicate than move to a util in the heat of the moment. But, I'll look at that refactor if this dupe bothers anyone.

File: packages/api-graphql/__tests__/internals/generateClient.test.ts (hunk 1 above)
Comment (svidgen) [reply packages/api-graphql/__tests__/internals/generateClient.test.ts]: Yeah. Completely accidental and unnecessary. Removed.

File: packages/api-graphql/__tests__/internals/implicit-auth-fields.test.ts (hunk 3 above)
Comment (iartemiev) [reply packages/api-graphql/__tests__/internals/implicit-auth-fields.test.ts:32]: This is the 3rd occurrence of this fn definition in the GQL tests, so I'd say let's move it to utils.

Comment (svidgen) [reply packages/api-graphql/__tests__/internals/implicit-auth-fields.test.ts:32]: Taking a look.

Comment (svidgen) [reply packages/api-graphql/__tests__/internals/implicit-auth-fields.test.ts]: OK. That wasn't so bad. Finished consolidating a few other bits as well.

**Review by iartemiev — APPROVED** [review]

**Review by stocaaro — APPROVED** [review]

**Pull request merged**